internal/handler/web/users.go

```diff
@@ -89,3 +89,31 @@ func (h *Handler) PostUsersApprovedResearchersImportCsv(ctx *gin.Context) {
 	}
 	ctx.Status(http.StatusNoContent)
 }
+
+func (h *Handler) PostUsersInvite(ctx *gin.Context) {
+	var invite openapi.PostUsersInviteJSONRequestBody
+	if err := bindJSONOrSetError(ctx, &invite); err != nil {
+		return
+	}
+
+	user := middleware.GetUser(ctx)
+	attributes, err := h.users.Attributes(user)
+	if err != nil {
+		setError(ctx, err, "Failed to get user attributes")
+		return
+	}
+
+	sponsor := types.Sponsor{
+		Username:   user.Username,
+		ChosenName: attributes.ChosenName,
+	}
+
+	// todo: check if invitee already exists in our db
+
+	if err := h.users.InviteUser(ctx, invite.Email, sponsor); err != nil {
+		setError(ctx, err, "Failed to send invite")
+		return
+	}
+
+	ctx.Status(http.StatusNoContent)
```

Comment:
This might be one more for @t-young31 . Does this need to pass some sort of result to the user? At the moment if a server error is caught the user won't be notified about it. So even if a `500` is returned the UI will still say `invitation sent` and it looks like the invite was sent ok.

I think the error would need to be handled at least on the frontend. But possibly also caught on the backend and sent as a `result`. BUt I'll defer to @t-young31 here for more guidance.

Reply:
I guess depends what we want to show the user. Just the response code could be enough – e.g. `204` -> 'Invite sent', `406` -> 'Sorry your request wasn't valid', `500` -> 'Sorry something went wrong. Please try again'